### PR TITLE
py-httpx: added version 0.27.0 and 0.27.2

### PR DESCRIPTION
**Pull request by qwertos**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-httpx/package.py [hunk 1]
```diff
@@ -11,37 +11,49 @@ class PyHttpx(PythonPackage):
...
+
+        depends_on("py-httpcore@0.11.0:0.11", when="@0.15.2")
+        depends_on("py-httpcore@0.14.5:0.14", when="@0.22")
+        depends_on("py-httpcore@0.15:0.16", when="@0.23")
+        depends_on("py-httpcore@1", when="@0.27:")
+
+        depends_on("py-anyio", when="@0.27:")
+        depends_on("py-idna", when="@0.27:")
+
+        depends_on("py-sniffio@1.0:1", when="@0.11.1")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/py-httpx/package.py:43]: It's actually sufficient to say `py-sniffio@1` (see the `python` example at https://spack.readthedocs.io/en/latest/packaging_guide.html#version-selection).

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/py-httpx/package.py:43]: That applies here and below for `py-h2` and others.

File: var/spack/repos/builtin/packages/py-httpx/package.py [hunk 2]
```diff
@@ -11,37 +11,49 @@ class PyHttpx(PythonPackage):
...
+    version("0.27.2", sha256="f7c2be1d2f3c3c3160d441802406b206c2b76f5947b11115e6df10c6c65e66c2")
+    version("0.27.0", sha256="a0cb88a46f32dc874e04ee956e4c2764aba2aa228f650b06788ba6bda2962ab5")
     version("0.23.3", sha256="9818458eb565bb54898ccb9b8b251a28785dd4a55afbc23d0eb410754fe7d0f9")
     version("0.22.0", sha256="d8e778f76d9bbd46af49e7f062467e3157a5a3d2ae4876a4bbfd8a51ed9c9cb4")
     version("0.15.2", sha256="713a2deaf96d85bbd4a1fbdf0edb27d6b4ee2c9aaeda8433042367e4b9e1628d")
     version("0.11.1", sha256="7d2bfb726eeed717953d15dddb22da9c2fcf48a4d70ba1456aa0a7faeda33cf7")
 
     variant("http2", default=False, when="@0.15.2:", description="Enable http2 support")
 
+    depends_on("python@3.8", when="@0.27:", type=("build", "run"))
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/py-httpx/package.py:27]: Curious.  Why not include this after the `default_args` context below?

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/py-httpx/package.py:27]: Also, shouldn't this be `python@3.8:` (based on the two newer `pyproject.toml` files)?

File: var/spack/repos/builtin/packages/py-httpx/package.py [hunk 3]
```diff
@@ -11,37 +11,49 @@ class PyHttpx(PythonPackage):
...
+    with default_args(type=("build", "run")):
+        depends_on("py-certifi")
+
+        depends_on("py-httpcore@0.11.0:0.11", when="@0.15.2")
+        depends_on("py-httpcore@0.14.5:0.14", when="@0.22")
+        depends_on("py-httpcore@0.15:0.16", when="@0.23")
+        depends_on("py-httpcore@1", when="@0.27:")
+
+        depends_on("py-anyio", when="@0.27:")
+        depends_on("py-idna", when="@0.27:")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/py-httpx/package.py:41]: Where are these coming from?  I don't see them in `pyproject.toml` when I stage `0.27.0` and `0.27.2`.

**Review by tldahlgren — COMMENTED** [review]
Confirmed the two new version sha256s.  I have some questions and suggestions related to the dependencies.

**Review by tldahlgren — CHANGES_REQUESTED** [review]
See previous comment.

File: var/spack/repos/builtin/packages/py-httpx/package.py (hunk 3 above)
Comment (qwertos) [reply var/spack/repos/builtin/packages/py-httpx/package.py:41]: From the pypi tarball, file `pyproject.toml` lines 33 and 34.

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/py-httpx/package.py:41]: 🤦

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**